mono_repo/lib/src/root_config.dart

```diff
@@ -116,4 +135,42 @@ class RootConfig extends ListBase<PackageConfig> {
   @override
   void operator []=(int index, PackageConfig pkg) =>
       throw UnsupportedError('This List is read-only.');
+
+  /// Parse any github action versions from a workflow file.
+  ///
+  /// This returns a map of <action name> to <action version>.
+  static Map<String, String> parseActionVersions(File workflowFile) {
+    // "dart-lang/setup-dart@6a218f2413a3e78e9087f638a238f6b40893203d"
+    final usageRegex = RegExp(r'([\w-]+)\/([\w-]+)@([\w\.]+)');
```

Comment:
I believe that `_` and `.` are both valid repo/org names also

Reply:
Oh I guess those are considered word characters?

Reply:
Nevermind, `.` is not a "word character" so we should add that one

Reply:
Gotcha - added `\.` to the 1st and 2nd regex group.